Mathlib/Algebra/Module/Submodule/Pointwise.lean

```diff
@@ -277,4 +289,221 @@ scoped[Pointwise] attribute [instance] Submodule.pointwiseMulActionWithZero
 
 end
 
+section set_acting_on_submodules
+
+variable [AddCommMonoid M] [Module R M]
+
+/--
+let `S ⊆ R` be a set and `N ≤ M` be a submodule, then `S • N` is the smallest submodule containing
+all `r • n` where `r ∈ S` and `n ∈ N`.
+-/
+protected def pointwiseSetSMulSubmodule : SMul (Set R) (Submodule R M) where
+  smul s N := sInf { p | ∀ ⦃r : R⦄ ⦃n : M⦄, r ∈ s → n ∈ N → r • n ∈ p }
+
+scoped[Pointwise] attribute [instance] Submodule.pointwiseSetSMulSubmodule
+
+variable (s : Set R) (N : Submodule R M)
+
+lemma mem_set_smul_submodule_def (x : M) :
+    x ∈ s • N ↔
+  x ∈ sInf { p : Submodule R M | ∀ ⦃r : R⦄ {n : M}, r ∈ s → n ∈ N → r • n ∈ p } := Iff.rfl
+
+lemma mem_set_smul_submodule_of_mem_mem {r : R} {m : M} (mem1 : r ∈ s) (mem2 : m ∈ N) :
+    r • m ∈ s • N := by
+  rw [mem_set_smul_submodule_def, mem_sInf]
+  exact fun _ h => h mem1 mem2
+
+lemma set_smul_submodule_le (p : Submodule R M)
+    (closed_under_smul : ∀ ⦃r : R⦄ ⦃n : M⦄, r ∈ s → n ∈ N → r • n ∈ p) :
+    s • N ≤ p :=
+  sInf_le closed_under_smul
+
+lemma set_smul_submodule_eq_of_le (p : Submodule R M)
+    (closed_under_smul : ∀ ⦃r : R⦄ ⦃n : M⦄, r ∈ s → n ∈ N → r • n ∈ p)
+    (le : p ≤ s • N) :
+    s • N = p :=
+  le_antisymm (set_smul_submodule_le s N p closed_under_smul) le
+
+lemma set_smul_submodule_mono_right {p q : Submodule R M} (le : p ≤ q) :
+    s • p ≤ s • q :=
+  set_smul_submodule_le _ _ _ fun _ _ hr hm => mem_set_smul_submodule_of_mem_mem (mem1 := hr)
+    (mem2 := le hm)
+
+lemma set_smul_submodule_mono_left {s t : Set R} (le : s ≤ t) :
+    s • N ≤ t • N :=
+  set_smul_submodule_le _ _ _ fun _ _ hr hm => mem_set_smul_submodule_of_mem_mem (mem1 := le hr)
+    (mem2 := hm)
+
```

Comment:
I think there should be a `s • p ≤ t • q` lemma for `s ≤ t` and `p ≤ q`.

Reply:
see `set_smul_submodule_le_of_le_le`